inference-engine/src/mkldnn_plugin/nodes/mkldnn_deconv_node.cpp

```diff
@@ -334,94 +407,300 @@ bool MKLDNNDeconvolutionNode::created() const {
     return getType() == Deconvolution;
 }
 
-void MKLDNNDeconvolutionNode::createPrimitive() {
-    if (prim)
-        return;
+bool MKLDNNDeconvolutionNode::needShapeInfer() const {
+    if (inputShapesModified()) {
+        return true;
+    }
+    if (externOutShape) {
+        if (currentOutSpatialDims != get3rdInputData()) {
+            return true;
+        }
+    }
 
-    if (isInt8) {
-        auto prim_desc = createPrimitiveDescriptor<deconvolution_forward::primitive_desc,
-                deconvolution_forward::desc>(attr);
+    return false;
+}
+
+std::vector<VectorDims> MKLDNNDeconvolutionNode::shapeInfer() const {
+    const auto &dataMemPtr = getParentEdgesAtPort(0)[0]->getMemoryPtr();
+    std::vector<int32_t> outSpDims;
+    if (externOutShape) {
+        outSpDims = get3rdInputData();
+    }
+    return {shapeInferInternal(dataMemPtr->getStaticDims(), outSpDims)};
+}
+
+VectorDims MKLDNNDeconvolutionNode::shapeInferInternal(const VectorDims &inDims, std::vector<int32_t> outSpDims) const {
+    std::vector<ov::StaticShape> inputShapes = {
+            inDims,
+            getWeightDims()
+    };
+
+    std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>> inputValues;
+
+    if (getParentEdges().size() > 2) {
+        if (outSpDims.size() != getInputShapeAtPort(2).getStaticDims()[0]) {
```

Comment:
Shouldn't this check be part of isSupportedOperation?

Reply:
I agree, but this in not the single place your are using that data. I guess the check can be part of readOutputSpatialDims.

Reply:
In `readOutputSpatialDims` I just read `shapeMemPtr->getStaticDims()[0]` elements from third input, because we suppose that third input shape is static, so we no need to validate number of elements inside `readOutputSpatialDims`.
I validate `outSpDims` here and in other function in case if we passed them as function's argument.

I extended check in ctor that third input shape is static and equal input spatial dims number.